modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala

```diff
@@ -835,77 +836,93 @@ object Publish extends ScalaCommand[PublishOptions] with BuildCommandHelpers {
         }
     }
 
-    val signerOpt = publishOptions.contextual(isCi).signer.orElse {
+    val signerKind = publishOptions.contextual(isCi).signer.getOrElse {
```

Comment:
NIT I get the feeling that using a `match` clause on `publishOptions.contextual(isCi)` rather than the nested if clauses could significantly improve the readability of this part

Reply:
I think that will just replace the getOrElse, so there's no gain from that. Also I've flattened the if clause so it's a little more readable now.